sql/src/main/java/org/apache/druid/sql/calcite/rel/DruidQuery.java

```diff
@@ -1448,15 +1449,6 @@ private ScanQuery toScanQuery()
         );
         return null;
       }
-      if (!dataSource.isConcrete()) {
```

Comment:
This was a change in #12418 that was reverted. Since we are using a table data source and not a query data source now the isConcrete() check is not needed and can be removed